.githooks/pre-push

```diff
@@ -1,31 +1,38 @@
 #!/bin/sh
 
-# An example hook script to verify what is about to be pushed.  Called by "git
-# push" after it has checked the remote status, but before anything has been
-# pushed.  If this script exits with a non-zero status nothing will be pushed.
-#
-# This hook is called with the following parameters:
-#
-# $1 -- Name of the remote to which the push is being done
-# $2 -- URL to which the push is being done
-#
-# If pushing without using a named remote those arguments will be equal.
-#
-# Information about the commits which are being pushed is supplied as lines to
-# the standard input in the form:
-#
-#   <local ref> <local oid> <remote ref> <remote oid>
+# Pre-push hook script to run code quality checks and ensure consistency.
 #
+# This script will execute two custom scripts located in the `scripts/` directory:
+# 1. Enforce cargo version 1.75.
+# 2. clippy-on-all-workspaces.sh: Runs Clippy, tests, and formatting on all specified workspaces.
+# 3. sv2-header-check.sh: Ensures the `sv2.h` file generated by `build_header.sh` matches the
+#    committed version.
 
+# Exit immediately if any command exits with a non-zero status and print each command before
+# executing it.
 set -xe
 
-remote="$1"
-url="$2"
+# Enforce cargo version 1.75
+REQUIRED_CARGO_VERSION="1.75.0"
+INSTALLED_CARGO_VERSION=$(cargo --version | awk '{print $2}')
+
+if [ "$INSTALLED_CARGO_VERSION" != "$REQUIRED_CARGO_VERSION" ]; then
+    echo "Error: Cargo version $REQUIRED_CARGO_VERSION is required. Installed version is $INSTALLED_CARGO_VERSION."
+    exit 1
+fi
```

Comment:
@Fi3, regarding your comment [here](https://github.com/stratum-mining/stratum/pull/1039#discussion_r1671966258),  do you like how I am enforcing the `cargo` version? Open to alternatives

Reply:
I would just try to run the wanted cargo verision `cargo +1.75.0 ...` and fail if not avaible, the error from cargo is already descriptive. Instead of 1.75.0 you use REQUIRED_CARGO_VERSION

Reply:
I got this error:
```
+ REQUIRED_CARGO_VERSION=1.75.0
++ cargo --version
++ awk '{print $2}'
+ INSTALLED_CARGO_VERSION=1.77.2
+ '[' 1.77.2 '!=' 1.75.0 ']'
+ echo 'Error: Cargo version 1.75.0 is required. Installed version is 1.77.2.'
Error: Cargo version 1.75.0 is required. Installed version is 1.77.2.
+ exit 1
```

I think this should only fail if the installed cargo version is smaller than MSRV, not bigger.

Reply:
Update the logic to enforce `1.75.0` as minimum. Checked that `1.74.0` cargo fails this check and that `1.75.0` and `1.77.0` cargo passes this check.